### ISSUE TITLE
AWS099 False positive for actions without required resource specification

### ISSUE THREAD

**Issue by ph-l** [aquasecurity/tfsec#915]
**Describe the bug**
AWS099 - IAM policy should avoid use of wildcards and instead apply the principle of least privilege

From https://docs.aws.amazon.com/service-authorization/latest/reference/list_identityandaccessmanagement.html:
> If there is no value for this column, you must specify all resources ("*") in the Resource element of your policy statement. 

**To Reproduce**
```terraform
resource "aws_iam_policy" "policy" {
  name        = "mfa-policy"

  policy = <<EOT
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": [
        "iam:ListVirtualMFADevices"
      ],
      "Effect": "Allow",
      "Resource": "*"
    }
  ]

}
EOT
}
```

**Additional context**
Fails with `guardduty:AcceptInvitation` action and `*` as resource too, did not test with other services.

**Comment by liamg** [comment]
Thanks, will go through the table and fix for everything it covers :+1:

**Comment by liamg** [comment]
Writing a scraper for the docs to grab all allowed actions - hopefully will have this sorted today/tomorrow.

**Comment by liamg** [comment]
Fixed in v0.50.6

**Comment by liamg** [comment]
The list of actions for which a wildcard resource is allowed/required is here for future reference: https://gist.github.com/liamg/eafcb27f29fd859e80ff871eaac11bb5
